### PR TITLE
split env vars using apiDefSchema

### PR DESCRIPTION
**Pull request by msadovyi**

### CODE REVIEW

**Review by germankrause — APPROVED** [review]

File: packages/config/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const deepSearch = (
```
Comment (valcker) [inline packages/config/src/utils.ts:1]: Add a jsdoc comment with an explanation of what the method is doing and it's params.

File: packages/config/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  object: { [key: string]: any },
+  key: string,
+  predicate: (k: string, v: any) => boolean
+): any => {
```
Comment (valcker) [inline packages/config/src/utils.ts:5]: Can we get rid of _any_ here?

File: packages/config/src/use-env.ts [hunk 3]
```diff
@@ -1,10 +1,21 @@
+import { deepSearch } from './utils';
+
 const envVarRegExp = /^@@/;
 
-export default function useEnv(config: { [key: string]: any }) {
+const isArrayProperty = (key: string, schema: { [key: string]: any }): boolean => {
```
Comment (valcker) [inline packages/config/src/use-env.ts:5]: It should be _isStringArrayProperty_ then, if we are checking for 'array' and 'string' types.

File: packages/config/src/use-env.ts [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
+  return deepSearch(schema, key, (k, v) => v?.type === 'array' && v?.items?.type === 'string');
+};
+
+export default function useEnv(config: { [key: string]: any }, schema?: { [key: string]: any }): void {
```
Comment (valcker) [inline packages/config/src/use-env.ts:9]: Do we have tests for useEnv?

File: packages/config/src/utils.ts (hunk 1 above)
Comment (valcker) [inline packages/config/src/utils.ts:1]: Add tests

**Review by valcker — APPROVED** [review]

File: packages/config/src/utils.ts (hunk 2 above)
Comment (msadovyi) [reply packages/config/src/utils.ts:26]: It returns value and it can be anything.

**Pull request merged**